### PR TITLE
:bug: HCL AppScan: Improve Robustness

### PR DESCRIPTION
**Pull request by manuel-sommer**
issue #9279

### CODE REVIEW

**Comment by manuel-sommer** [comment]
Could you take a look and review this also please @mtesauro ?

**Review by mtesauro — APPROVED** [review]
Approved

**Review by Maffooch — APPROVED** [review]

File: dojo/tools/hcl_appscan/parser.py [hunk 1]
```diff
@@ -24,60 +33,85 @@ def get_findings(self, file, test):
...
                         case 'location':
-                            location = item.text
-                            description = description + "Location: " + location + "\n"
+                            location = self.xmltreehelper(item)
+                            description = description + "Location:" + location + "\n"
                         case 'domain':
-                            domain = item.text
-                            description = description + "Domain: " + domain + "\n"
+                            domain = self.xmltreehelper(item)
+                            title += "_" + domain
```
Comment (cneill) [inline dojo/tools/hcl_appscan/parser.py:60]: I don't see much benefit to introducing spaces into the title in this way, given that the individual parts are already separated by `_`. A `.strip()` seems appropriate here

```suggestion
                            title += "_" + domain.strip()
```

File: dojo/tools/hcl_appscan/parser.py [hunk 2]
```diff
@@ -24,60 +33,85 @@ def get_findings(self, file, test):
...
+                        case 'entity':
+                            entity = self.xmltreehelper(item)
+                            title += "_" + entity
+                            description = description + "Entity:" + entity + "\n"
+                        case 'security-risks':
+                            security_risks = self.xmltreehelper(item)
+                            description = description + "Security-Risks:" + security_risks + "\n"
+                        case 'cause-id':
+                            causeid = self.xmltreehelper(item)
+                            title += "_" + causeid
```
Comment (cneill) [inline dojo/tools/hcl_appscan/parser.py:74]: Same comment as above

```suggestion
                            title += "_" + causeid.strip()
```

File: dojo/tools/hcl_appscan/parser.py [hunk 3]
```diff
@@ -24,60 +33,85 @@ def get_findings(self, file, test):
...
-                            description = description + "Domain: " + domain + "\n"
+                            domain = self.xmltreehelper(item)
+                            title += "_" + domain
+                            description = description + "Domain:" + domain + "\n"
+                        case 'threat-class':
+                            threatclass = self.xmltreehelper(item)
+                            description = description + "Threat-Class:" + threatclass + "\n"
+                        case 'entity':
+                            entity = self.xmltreehelper(item)
+                            title += "_" + entity
```
Comment (cneill) [inline dojo/tools/hcl_appscan/parser.py:67]: Same comment as above

```suggestion
                            title += "_" + entity.strip()
```

File: dojo/tools/hcl_appscan/parser.py [hunk 4]
```diff
@@ -24,60 +33,85 @@ def get_findings(self, file, test):
...
+                        case 'security-risks':
+                            security_risks = self.xmltreehelper(item)
+                            description = description + "Security-Risks:" + security_risks + "\n"
+                        case 'cause-id':
+                            causeid = self.xmltreehelper(item)
+                            title += "_" + causeid
+                            description = description + "Cause-Id:" + causeid + "\n"
+                        case 'url-name':
+                            urlname = self.xmltreehelper(item)
+                            title += "_" + urlname
```
Comment (cneill) [inline dojo/tools/hcl_appscan/parser.py:78]: Same comment as above

```suggestion
                            title += "_" + urlname.strip()
```

File: dojo/tools/hcl_appscan/parser.py [hunk 5]
```diff
@@ -24,60 +33,85 @@ def get_findings(self, file, test):
...
                         case 'element-type':
-                            elementtype = item.text
-                            description = description + "ElementType: " + elementtype + "\n"
+                            elementtype = self.xmltreehelper(item)
+                            description = description + "ElementType:" + elementtype + "\n"
                         case 'path':
-                            path = item.text
-                            description = description + "Path: " + path + "\n"
+                            path = self.xmltreehelper(item)
+                            title += "_" + path
```
Comment (cneill) [inline dojo/tools/hcl_appscan/parser.py:88]: Same comment as above

```suggestion
                            title += "_" + path.strip()
```

File: unittests/tools/test_hcl_appscan_parser.py [hunk 6]
```diff
@@ -17,8 +17,23 @@ def test_many_findings(self):
         findings = parser.get_findings(my_file_handle, None)
         my_file_handle.close()
         self.assertEqual(60, len(findings))
-        self.assertEqual(findings[0].title, "Unencrypted Login Request_mani-virtual-machine_/dvja-1.0-SNAPSHOT/register.action")
-        self.assertEqual(findings[1].title, "Unencrypted Login Request_mani-virtual-machine_/dvja-1.0-SNAPSHOT/login.action;jsessionid=AD12F9CF7835CC92885A381859462BAC")
+        self.assertEqual(findings[0].title, " Unencrypted Login Request_ mani-virtual-machine_ /dvja-1.0-SNAPSHOT/register.action")
+        self.assertEqual(findings[1].title, " Unencrypted Login Request_ mani-virtual-machine_ /dvja-1.0-SNAPSHOT/login.action;jsessionid=AD12F9CF7835CC92885A381859462BAC")
```
Comment (cneill) [inline unittests/tools/test_hcl_appscan_parser.py:21]:
```suggestion
        self.assertEqual(findings[0].title, " Unencrypted Login Request_mani-virtual-machine_/dvja-1.0-SNAPSHOT/register.action")
        self.assertEqual(findings[1].title, " Unencrypted Login Request_mani-virtual-machine_/dvja-1.0-SNAPSHOT/login.action;jsessionid=AD12F9CF7835CC92885A381859462BAC")
```

File: unittests/tools/test_hcl_appscan_parser.py [hunk 7]
```diff
@@ -17,8 +17,23 @@ def test_many_findings(self):
...
+        self.assertEqual(findings[9].cwe, 522)
+
+    def test_issue_9279(self):
+        my_file_handle = open("unittests/scans/hcl_appscan/issue_9279.xml")
+        parser = HCLAppScanParser()
+        findings = parser.get_findings(my_file_handle, None)
+        my_file_handle.close()
+        self.assertEqual(18, len(findings))
+        self.assertEqual(findings[0].title, " attUnnecessaryResponseHeaders_ 7089695691196187648_ insecureWebAppConfiguration")
+        self.assertEqual(findings[1].title, " attHttpsToHttp_ 7089695691196187648_ sensitiveDataNotSSL")
```
Comment (cneill) [inline unittests/tools/test_hcl_appscan_parser.py:35]:
```suggestion
        self.assertEqual(findings[0].title, " attUnnecessaryResponseHeaders_7089695691196187648_insecureWebAppConfiguration")
        self.assertEqual(findings[1].title, " attHttpsToHttp_7089695691196187648_sensitiveDataNotSSL")
```

**Review by cneill — CHANGES_REQUESTED** [review]

**Review by cneill — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Pull request merged**